plugins/web/opentelemetry-plugin-user-interaction/test/userInteraction.nozone.test.ts

```diff
@@ -89,6 +89,40 @@ describe('UserInteractionPlugin', () => {
       context.disable();
     });
 
+    it('should not break removeEventListener', () => {
+      let called = false;
+      const listener = function () {
+        called = true;
+      };
+      // add same listener three different ways
```

Comment:
I'm curious about one scenario, to check that can you modify this a bit to test the case with overwriting the previous event with different callback
```
let called = false;
let called2 = false;
const listener = function () {
  called = true;
};
const listener2 = function () {
  called2 = true;
};
document.body.addEventListener('bodyEvent', listener);
document.body.addEventListener('bodyEvent', listener2);
document.body.addEventListener('bodyEvent2', listener);

// now fun
document.body.dispatchEvent(new Event('bodyEvent'));
assert.strictEqual(called, false);
assert.strictEqual(called2, true);
```

Reply:
fixed my example

Reply:
Yep, good catch and an easy fix for this one based on the tracking state.  Pushing fix momentarily.

Reply:
There is one more thing a bonus :). Currently you can call `addEventListener` providing a 3rd param `options`. In options you can set `once:true` which means the event will be removed automatically. It might be worth to test that too.

Reply:
```
let called2 = 0;
const listener2 = function () {
  called2++;
};
document.body.addEventListener('bodyEvent', listener2);
document.body.addEventListener('bodyEvent', listener2, { once: true });

// now fun
document.body.dispatchEvent(new Event('bodyEvent'));
assert.strictEqual(called2, 1);
document.body.dispatchEvent(new Event('bodyEvent'));
assert.strictEqual(called2, 2); // error ?
```

Reply:
and you can revert it add `once` first and then add without options, heh small things and many edge cases

Reply:
Argh.  There's also the case that once:true shouldn't prevent us from allowing it to be adding later (it's not really a double-register).  Will dive in and fix this one tomorrow.  Thanks!

Reply:
thx for taking this into deep details :)

Reply:
Couldn't leave it alone; handled the `once:true` callbacks properly being removed when an event is dispatched.